packages/react-router-dom/modules/Link.js

```diff
@@ -52,9 +57,24 @@ class Link extends React.Component {
     }
   }
 
-  render() {
-    const { replace, to, ...props } = this.props // eslint-disable-line no-unused-vars
+  absolute() {
+    const { to } = this.props
+    const { match } = this.context.router
+    const base = (match && match.url) ? match.url : ''
+    return resolveLocation(to, base)
+  }
+
+  componentWillMount() {
+    this.unlisten = this.context.router.listen(() => { this.forceUpdate() })
```

Comment:
Sadly, I realized that I had a bug and a `location` prop is actually necessary for a `<Switch>` to work with transitions (the leaving component still calls `cWRP`). Besides that, I'm really liking that branch.

Also, it is true. I do have a [face](https://cloud.githubusercontent.com/assets/1127037/23105272/365ea602-f6a2-11e6-9494-1908b83453c8.png).